src/NovelAudioService.cpp

```diff
@@ -0,0 +1,367 @@
+  }
+  else
+  {
+    if (Mix_Playing(_channelMap[soundName]) == 1)
+    {
+      std::cout << "Already playing on channel " << _channelMap[soundName] << std::endl;
+      return;
+    }
+
+    Mix_PlayChannel(_channelMap[soundName], _sounds[soundName], 0);
+  }
+
+}
+
+void NovelAudioService::playSound(std::string soundName, int loops) {
+  auto existingSound = _sounds.find(soundName);
+  if (existingSound == _sounds.end())
+  {
+    loadSound(soundName);
+    existingSound = _sounds.find(soundName);
+    if (existingSound == _sounds.end())
+    {
+      return;
+    }
+  }
+
+  auto onChannel = _channelMap.find(soundName);
+  if (onChannel == _channelMap.end())
+  {
+    _channelMap[soundName] = _nextChannel;
+    Mix_PlayChannel(_nextChannel, _sounds[soundName], loops-1);
+    int nextChannelTest = _nextChannel + 1;
+    _nextChannel = (nextChannelTest >= 2 || nextChannelTest < 0) ? 0 : nextChannelTest;
+  }
+  else
+  {
+    if (Mix_Playing(_channelMap[soundName]) == 1)
+    {
+      std::cout << "Already playing on channel " << _channelMap[soundName] << std::endl;
+      return;
+    }
+    Mix_PlayChannel(_channelMap[soundName], _sounds[soundName], loops-1);
+  }
+
+}
+
+void NovelAudioService::stopSound(std::string soundName) {
+  Mix_HaltChannel(_channelMap[soundName]);
+}
+
+int NovelAudioService::getSoundChannel(std::string soundName) {
+  auto existingSound = _channelMap.find(soundName);
+  if (existingSound == _channelMap.end())
+  {
+    std::cerr << "ERROR: Cannot get channel for sound " << soundName << std::endl;
+    return -9;
+  }
+  return _channelMap[soundName];
+}
+
+void NovelAudioService::setSoundVolume(std::string soundName, float value) {
+  Mix_VolumeChunk(_sounds[soundName], convertToMixVolume(value));
+}
+
+void NovelAudioService::setSoundPosition(std::string soundName, int angle, int distance) {
+  if (_channelMap.find(soundName) != _channelMap.end() && _channelMap[soundName] != -1)
+  {
+    Mix_SetPosition(_channelMap[soundName], angle, distance);
+  }
+  else
+  {
+    std::cerr << soundName << " is not allocated to a channel already!" << std::endl;
+  }
+}
+
+void NovelAudioService::setSoundDistance(std::string soundName, int distance) {
+    if (_channelMap.find(soundName) != _channelMap.end() && _channelMap[soundName] != -1)
+  {
+    Mix_SetDistance(_channelMap[soundName], distance);
+  }
+  else
+  {
+    std::cerr << soundName << " is not allocated to a channel already!" << std::endl;
+  }
+}
+
+void NovelAudioService::setSoundPanning(std::string soundName, int leftChannelVolume, int rightChannelVolume) {
+    if (_channelMap.find(soundName) != _channelMap.end() && _channelMap[soundName] != -1)
+  {
+    Mix_SetPanning(_channelMap[soundName], leftChannelVolume, rightChannelVolume);
+  }
+  else
+  {
+    std::cerr << soundName << " is not allocated to a channel already!" << std::endl;
+  }
+}
+
+void NovelAudioService::loadMusic(std::string input) {
```

Comment:
What is the intended difference between `music` and `sound`?

Is this not something that could just be handled via an `enum` type or auto-detection; so you could have a single `load`/`play` API?

Reply:
Music and Sound are referenced differently in SDL_Mixer (i.e. `MixMusic` vs `MixChunk`), hence the separation here.

Would wrapping it in a `struct` (so `enum` type and file name) work to condense these?

Reply:
Right, but I think my question is: "does this difference need to be exposed to the user as part of the method to call".

It seems like taking a struct that defines the name, enum type, and any other info and then having the APIs internally lookup what the "type" is, so it knows which mixer API to call is better.

Reply:
-- This also allows easier use of other audio backends if you ever want to support them 😄

Reply:
(inb4 I'm asked about OpenAL in the future)
It shouldn't be exposed really, so I'll condense this in the next push

Reply:
Condensed the music and sound loading functions in the latest push